state/bulk.go

```diff
@@ -47,6 +57,31 @@ func NewDefaultBulkStore(base BaseStore) BulkStore {
 
 // BulkGet performs a Get operation in bulk.
 func (b *DefaultBulkStore) BulkGet(ctx context.Context, req []GetRequest, opts BulkGetOpts) ([]BulkGetResponse, error) {
+	return DoBulkGet(ctx, req, opts, b.base.Get)
+}
+
+// BulkSet performs a bulk save operation.
+func (b *DefaultBulkStore) BulkSet(ctx context.Context, req []SetRequest) error {
+	return b.BulkSetWithOptions(ctx, req, BulkStoreOpts{})
+}
+
+// BulkSetWithOptions performs a bulk save operation with options.
+func (b *DefaultBulkStore) BulkSetWithOptions(ctx context.Context, req []SetRequest, opts BulkStoreOpts) error {
+	return DoBulkSetDelete(ctx, req, b.base.Set, opts)
+}
+
+// BulkDelete performs a bulk delete operation.
+func (b *DefaultBulkStore) BulkDelete(ctx context.Context, req []DeleteRequest) error {
+	return b.BulkDeleteWithOptions(ctx, req, BulkStoreOpts{})
+}
+
+// BulkDeleteWithOptions performs a bulk delete operation with options
+func (b *DefaultBulkStore) BulkDeleteWithOptions(ctx context.Context, req []DeleteRequest, opts BulkStoreOpts) error {
+	return DoBulkSetDelete(ctx, req, b.base.Delete, opts)
+}
+
+// DoBulkGet performs BulkGet.
+func DoBulkGet(ctx context.Context, req []GetRequest, opts BulkGetOpts, getFn func(ctx context.Context, req *GetRequest) (*GetResponse, error)) ([]BulkGetResponse, error) {
```

Comment:
This method is exported so components that want to implement some Bulk* methods natively can pick-and-choose. Example is RethinkDB